src/api/app/views/webui/shared/_buildresult_box.html.haml

```diff
@@ -1,51 +1,69 @@
 :ruby
   index ||= ''
+  project ||= nil
+  package ||= nil
+  bs_request ||= nil
+  buildable = package || project
 
-  # For a request staged in a staging project, we display the Build Results / RPM Lint from the staging project instead
-  is_staged_request = defined?(bs_request) && !bs_request.staging_project_id.nil?
-  project = bs_request.staging_project.name if is_staged_request
+  if buildable
+    # For a request staged in a staging project, we display the Build Results / RPM Lint from the staging project instead
+    is_staged_request = bs_request && !bs_request.staging_project_id.nil?
+    project = bs_request.staging_project.name if is_staged_request
 
-  ajax_data = {}
-  ajax_data['project'] = h(project) if defined?(project)
-  ajax_data['package'] = h(package) if defined?(package)
-  ajax_data['index'] = h(index) if defined?(index)
+    ajax_data = {}
+    ajax_data['project'] = h(project) if project
```

Comment:
:thought_balloon: All this logic can go into a Presenter, doing so, you remove all the logic from the view

Reply:
I wouldn't even call this view if the package or project are missing. We can handle more things from the controller...

In fact, I started to refactor this code but it was taking a lot of time. So I decided to simplify and just solve the issue.

Reply:
in P2 we first try to solve the problem.